src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerRemoveGeneralOrder.java

```diff
@@ -0,0 +1,107 @@
+package uk.gov.hmcts.divorce.caseworker.event;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.sdk.api.CCDConfig;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.ccd.sdk.api.ConfigBuilder;
+import uk.gov.hmcts.ccd.sdk.api.callback.AboutToStartOrSubmitResponse;
+import uk.gov.hmcts.ccd.sdk.type.ListValue;
+import uk.gov.hmcts.divorce.common.ccd.PageBuilder;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.DivorceGeneralOrder;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.divorcecase.model.UserRole;
+import uk.gov.hmcts.divorce.document.DocumentIdProvider;
+import uk.gov.hmcts.divorce.document.DocumentRemovalService;
+import uk.gov.hmcts.divorce.document.model.DivorceDocument;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static uk.gov.hmcts.divorce.divorcecase.model.State.POST_SUBMISSION_STATES;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CASE_WORKER;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SUPER_USER;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE_DELETE;
+
+@Component
+@Slf4j
+public class CaseworkerRemoveGeneralOrder implements CCDConfig<CaseData, State, UserRole> {
+
+    @Autowired
+    private DocumentRemovalService documentRemovalService;
+
+    @Autowired
+    private DocumentIdProvider documentIdProvider;
+
+    public static final String CASEWORKER_REMOVE_GO = "caseworker-remove-general-order";
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        new PageBuilder(configBuilder
+            .event(CASEWORKER_REMOVE_GO)
+            .forStates(POST_SUBMISSION_STATES)
+            .name("Remove General Order")
+            .description("Remove General Order")
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .showEventNotes()
+            .grant(CREATE_READ_UPDATE_DELETE, SUPER_USER)
+            .grantHistoryOnly(CASE_WORKER))
+            .page("removeGeneralOrders")
```

Comment:
For the event history, Fubara might also be able to confirm who should see that. We make CaseworkerRemoveScannedDocument and CaseworkerCreateGeneralOrder visible to most roles:
https://github.com/hmcts/nfdiv-case-api/blob/edb768814677050c267714d9eef2601ba90a0c07/src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerCreateGeneralOrder.java#L73

Reply:
Checked with Fubara and Isabel that history should be available to super user and caseworker